### PR TITLE
[6.x] Enforce usage of collect() helper

### PR DESCRIPTION
**Pull request by lucasmichot**
`new Collection` and `Collection::make` have been replaced for some times already by `collect()`.

This PR enforces this choice for sake of consistancy.

### CODE REVIEW

File: src/Illuminate/Pagination/LengthAwarePaginator.php [hunk 1]
```diff
@@ -50,7 +49,7 @@ public function __construct($items, $total, $perPage, $currentPage = null, array
         $this->lastPage = max((int) ceil($total / $perPage), 1);
         $this->path = $this->path !== '/' ? rtrim($this->path, '/') : $this->path;
         $this->currentPage = $this->setCurrentPage($currentPage, $this->pageName);
-        $this->items = $items instanceof Collection ? $items : Collection::make($items);
+        $this->items = collect($items);
```
Comment (lucasmichot) [inline src/Illuminate/Pagination/LengthAwarePaginator.php]: This check is anyway done in `\Illuminate\Support\Traits\EnumeratesValues::getArrayableItems`

File: src/Illuminate/Pagination/Paginator.php [hunk 2]
```diff
@@ -65,7 +65,7 @@ protected function setCurrentPage($currentPage)
      */
     protected function setItems($items)
     {
-        $this->items = $items instanceof Collection ? $items : Collection::make($items);
+        $this->items = collect($items);
```
Comment (lucasmichot) [inline src/Illuminate/Pagination/Paginator.php]: Same here, this check is useless.

File: src/Illuminate/Routing/SortedMiddleware.php [hunk 3]
```diff
@@ -19,7 +19,7 @@ public function __construct(array $priorityMap, $middlewares)
             $middlewares = $middlewares->all();
         }
 
-        $this->items = $this->sortMiddleware($priorityMap, $middlewares);
+        parent::__construct($this->sortMiddleware($priorityMap, $middlewares));
```
Comment (lucasmichot) [inline src/Illuminate/Routing/SortedMiddleware.php]: The parent constructor call was missed here

**Comment by GrahamCampbell** [comment]
Hmmm. I'd probably prefer not to use the global helpers within the core code tbh.

**Comment by lucasmichot** [comment]
> Hmmm. I'd probably prefer not to use the global helpers within the core code tbh.

Well `collect()` is already used everywhere.

**Comment by antonkomarev** [comment]
Why do we need to use global helper methods? Recently we've got an issue when some of third party packages overwritten global helper methods and out app started to work in wrong way. I'd prefer to use classes with unique namespace instead of helpers.
Helper methods might be used in client app, but there is no reason to use it in framework core.

**Comment by andrewnclark** [comment]
Personal preference but I prefer to avoid these helpers functions if at all possible and @antonkomarev's issue with overwritten functions is too easy to come across. I'd prefer to see `collect()` removed in favor of the class instead.

**Comment by lucasmichot** [comment]
The purpose of this PR is not to decide which of collect or new Collection syntaxes to use. It's just to keep everything consistent.

**Pull request closed without merging**